model_analyzer/result/model_config_measurement.py

```diff
@@ -0,0 +1,283 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model_analyzer.constants import COMPARISON_SCORE_THRESHOLD
+from model_analyzer.constants import LOGGER_NAME
+
+from model_analyzer.record.record import RecordType
+
+from functools import total_ordering
+import logging
+
+logger = logging.getLogger(LOGGER_NAME)
+
+
+@total_ordering
+class ModelConfigMeasurement:
+    """
+    Encapsulates the set of metrics obtained from a single model config's
+    RunConfig run
+    """
+
+    def __init__(self, model_config_name, model_specific_pa_params,
+                 non_gpu_data):
+        """
+        model_config_name : string
+            The model config name that was used in the RunConfig
+        model_specific_pa_params: dict
+            Dictionary of PA parameters that can change between models
+            in a multi-model RunConfig
+        non_gpu_data : list of Records
+            Metrics that do not have a GPU UUID associated with them,
+            from either CPU or PA
+        """
+
+        self._model_config_name = model_config_name
+        self._model_specific_pa_params = model_specific_pa_params
+        self._non_gpu_data = non_gpu_data
+
+        self._non_gpu_data_from_tag = {
+            type(metric).tag: metric for metric in self._non_gpu_data
+        }
+
+    @classmethod
+    def from_dict(cls, model_config_measurement_dict):
+        model_config_measurement = ModelConfigMeasurement(None, {}, [])
+
+        model_config_measurement._model_config_name = model_config_measurement_dict[
+            '_model_config_name']
+        model_config_measurement._model_specific_pa_params = model_config_measurement_dict[
+            '_model_specific_pa_params']
+        model_config_measurement._non_gpu_data = model_config_measurement_dict[
+            '_non_gpu_data']
+        model_config_measurement._non_gpu_data_from_tag = model_config_measurement_dict[
+            '_non_gpu_data_from_tag']
+
+        return model_config_measurement
+
+    def set_metric_weighting(self, metric_objectives):
+        """
+        Sets the metric weighting for this measurement based
+        on the objectives
+        
+        Parameters
+        ----------
+        metric_objectives : dict of RecordTypes
+            keys are the metric types, and values are the relative importance
+            of the keys with respect to each other
+        """
+        self._metric_weights = {
```

Comment:
I do'nt see this defined in init. Also, are there default values? If not, are we requiring this function be called before calling other functions?  Also, should it be part of from_dict?